p2p/src/peer_manager/peerdb/mod.rs

```diff
@@ -131,11 +131,23 @@ impl<T: NetworkingService> PeerDb<T> {
 
     /// Checks if the given address is already connected.
     pub fn is_address_connected(&self, address: &T::Address) -> bool {
-        self.addresses.contains(address)
+        self.connected_addresses.contains(address)
     }
 
-    pub fn known_addresses(&self) -> Vec<PeerAddress> {
-        self.addresses.iter().map(TransportAddress::as_peer_address).collect()
+    pub fn random_known_addresses(&self, count: usize) -> Vec<T::Address> {
```

Comment:
Please explain what these methods do and why they exist in a little snippet of documentation

Reply:
Sure, done.

Reply:
Also the other method, please...

Reply:
Added, thanks.